### PR TITLE
UI - Document Upload - Same File Upload Logic

### PR DESCRIPTION
**Pull request by shaangill025**
*Issue:*

- https://github.com/bcgov/entity/issues/23064

*Description of changes:*
When file with same name and extension is uploaded then only the latest one is included. Follow up to #124 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-125-lfoojrr5.web.app

**Comment by dimak1** [comment]
@shaangill025 please check if this is actual business requirement to use only the last uploaded doc if all names are the same. Thanks!

File: strr-web/components/bcros/form-section/principal-residence/Form.vue [hunk 1]
```diff
@@ -224,7 +225,12 @@ const uploadFile = (file: FileList) => {
     fileError.value = tPrincipalResidence('fileTypeError')
   } else {
     fileError.value = null
-    formState.supportingDocuments.push(file[0])
+    const existingIndex = formState.supportingDocuments.findIndex(doc => doc.name === newFile.name)
+    if (existingIndex !== -1) {
+      formState.supportingDocuments.splice(existingIndex, 1, newFile)
```
Comment (dimak1) [inline strr-web/components/bcros/form-section/principal-residence/Form.vue:230]: We can simplify here and just set the `formState.supportingDocuments[existingIndex] = newFile`

**Review by dimak1 — APPROVED** [review]
Approved with one comment.

**Comment by dimak1** [comment]
@shaangill025 any comments for closing this?

**Comment by shaangill025** [comment]
For now [MVP], it was decided to allow upload of same supporting files.

**Comment by dimak1** [comment]
@shaangill025 ok good, glad we clarified. Thanks!

**Pull request closed without merging**